d2/src/main/java/com/linkedin/d2/balancer/simple/ClusterLoadBalancerSubscriber.java

```diff
@@ -77,6 +77,7 @@ protected void handlePut(final String listenTo, final ClusterProperties discover
     }
     else
     {
+      _log.warn("Received a null cluster properties for {}", listenTo);
```

Comment:
Last time we added logs around, they ended up being very noisy. Can you double check this isn't the case for this logger?

Reply:
this log should not be logged unless there is a bug. And if there is a bug, we would want to see this log and take actions. So I think this log should be a must-show. I can make it an error instead?

Reply:
I guess the Uri one could use a rate limiter in case of huge clusters having many uris. The service and cluster data won't change frequently anyway.

Reply:
A rate limit seems like a good idea. I agree we want to fail loudly, but we don't want to fail so loudly it breaks people's logs to the point where we're scrambling to put in a fix! This has happened a few times already